Content.Server/PneumaticCannon/PneumaticCannonSystem.cs

```diff
@@ -78,12 +78,15 @@ private void OnShoot(Entity<PneumaticCannonComponent> cannon, ref GunShotEvent a
         if (gas == null && component.GasUsage > 0f)
             return;
 
-        if(TryComp<StatusEffectsComponent>(args.User, out var status)
-           && component.Power == PneumaticCannonPower.High)
+        if (args.User != null)
         {
-            _stun.TryParalyze(args.User, TimeSpan.FromSeconds(component.HighPowerStunTime), true, status);
-            Popup.PopupEntity(Loc.GetString("pneumatic-cannon-component-power-stun",
-                ("cannon", uid)), cannon, args.User);
+            if (TryComp<StatusEffectsComponent>(args.User, out var status)
+                && component.Power == PneumaticCannonPower.High)
+            {
+                _stun.TryParalyze(args.User.Value, TimeSpan.FromSeconds(component.HighPowerStunTime), true, status);
+                Popup.PopupEntity(Loc.GetString("pneumatic-cannon-component-power-stun",
+                    ("cannon", uid)), cannon, args.User.Value);
+            }
         }
```

Comment:
As elsewhere you're hardcoding in a bunch of specific functionality for ship guns which should be on their own system and not affecting the base systems.

Reply:
I'm not modifying the base system for ship cannons. I'm adding to it the ability for guns to fire without the user. This can be used in a wide range of ways

including: hacking weapons in hand, remote APE activation (EmitterComponent is horrible, I want to remove it), redesign of turrets

https://github.com/space-wizards/space-station-14/assets/96445749/c87f0f1d-6aa0-4c6b-be03-0085a481a607